### PR TITLE
feat: add amazon tam for react-advertising

### PR DESCRIPTION
**Pull request by kennylucv**
The main intention of this PR is to implement **Amazon TAM** as a new provider. 

Test coverage for Amazon TAM (APS):
- `.setup` method with errors ✅
- `.setup` method without errors ✅
- `.activate` method with errors ✅
- `.activate` method with errors ✅

### CODE REVIEW

File: package.json
Comment (jdw1996) [inline package.json]: I think this will be updated automatically when we publish the new version.

File: src/Advertising.js
Comment (jdw1996) [inline src/Advertising.js:85]: Why aren't we doing `Advertising.queueForGPT` here anymore? I can't remember what the problem was.

File: src/Advertising.js
Comment (jdw1996) [inline src/Advertising.js:157]: Same question about `queueForGPT` here

File: src/Advertising.test.js
Comment (jdw1996) [inline src/Advertising.test.js:141]: @jmanrumartinez I think we should add a test case where `activate` is called after `setup`, to make sure that `fetchBids` and `setDisplayBids` are called then. I don't believe that's covered by any of the above tests.

File: src/Advertising.test.js
Comment (jdw1996) [inline src/Advertising.test.js:866]: Note to reviewers: This test case name used to be split across two lines, as two strings concatenated with `+`. However, we were unable to run it in VS code until we changed it to be one string with no concatenation. I don't know why it would be a problem, but if anyone has a better solution, please let us know!

File: src/Advertising.test.js
Comment (jmanrumartinez) [reply src/Advertising.test.js:141]: Done! You're right! I was confused if there's already a test for it, but the above one it's just to test the `.setup` function! Thanks 🙌

File: package.json
Comment (jmanrumartinez) [reply package.json]: Then should we remove it before merging it right? To not cause any conflict.

File: src/Advertising.js
Comment (pahund) [reply src/Advertising.js:85]: the refreshSlots method does queueForGPT, so this should be OK

**Review by kennylucv — COMMENTED** [review]
Looks good to me @jmanrumartinez, I can't approve it since I've "opened" this PR

**Review by jmanrumartinez — APPROVED** [review]

**Comment by pahund** [comment]
Hey guyz, ready to merge this sucker?

**Review by pahund — APPROVED** [review]

**Comment by pahund** [comment]
The change is now available in the latest beta version [react-advertising@4.2.0-beta.29](https://www.npmjs.com/package/react-advertising/v/4.2.0-beta.29). We plan to release the final version 4.2.0 in October.

**Pull request merged**
